ykcompile/src/lib.rs

```diff
@@ -1331,4 +1427,90 @@ mod tests {
               leave
               ...", &tir_trace);
     }
+
+    fn ref_deref() -> u64 {
+        let mut x = 9;
+        let y = &mut x;
+        *y = 10;
+        let z = *y;
+        z
+    }
+
+    #[test]
+    fn test_ref_deref() {
+        let mut inputs = trace_inputs((0,));
+        let th = start_tracing(Some(TracingKind::HardwareTracing));
+        inputs.0 = ref_deref();
+        let sir_trace = th.stop_tracing().unwrap();
+        let tir_trace = TirTrace::new(&*sir_trace).unwrap();
+        let ct = TraceCompiler::<&(u64,)>::compile(tir_trace);
+        let mut args = (0,);
```

Comment:
Let's get into the habit of trying different args in each test case. I think it'll reduce "works by luck" test passes.

Reply:
It depends what you do with the args. If you overwrite it it won't matter, but if the output depends on it, we can't just randomly choose an input value.

Reply:
Ah, I see. The "input" here is actually only an output.

Reply:
Yes.